### ISSUE TITLE
Bug: [member-ordering] weird error "Member should be declared before all public instance get definitions"

### ISSUE THREAD

**Issue by KikoCosmetics** [typescript-eslint/typescript-eslint#5011]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.6.2&sourceType=module&code=KYDwDg9gTgLgBAYwDYEMDOa4EExgMIQB2MUESSwUcA3gLABQDDcLcaMKMAlgnAObB4AIQDyIgBQBKAFxsSXQnxrNWqqIICuUQnABEAIwgRdAbhUsAvk3qqwULgDdOwOZx79BcAPpCsAJSlZdntFZRtVVnUYLR0DFChTczgrRnCWO0dnV25eAXgvAC1-QLkQpTo0iKiYvQAveMTKlKS0T3148UIg+UVJMIjWGAALLjQAOi8AMyMQOABeOEIzJutVPLh2qBLghXKktU1tOGHRiemIEGXVZsrW+HPO7rK+ioGWE-Gpmfm9MA19JA8XRJG5rTwPGSlXb9N7VI4fM4zK6sUGsDJOGAuO7eB5dKG9GEDBFfC4-XTo5yNa6rNH2DEudYk7Y9PaVA7ReEjT7nS4gmnpOlZbFePhGR74vgvfaDLmI0kLcmCzFUlH8uAUzEefKiiDMsqEqqHHTEnnIyxqjUuEkgJ7QhWNJIIIjBDQIGDQKQG1EsPIACWA5F1kJ2oVesKNen9gZV5voKSAA&rules=N4IgAgLgngDgpgZwMYCcCWMIFpEBs0B2EA9IRHCgGYCGScWB1AtvTCnJWgB4gBcIAe0qUQAGnDR4ydJhwJ8RYnC4x8SNNkoBXAkghoBBLOwhaURyXD4gA7tXNiJsRKgzY8ZJSrUasTAQAmWrj0AEYCOgH2UFiWCNZCIuKQztJucgokBAI43mjq2NQEUNZ2DslxrrIeitlYhJQUKNShIbGpCcKOKVJV7vKeLEyhFFgCKAEUhADmfADaIE3jYqCTNMEQfKDjkyjW1LgwABYtcPpIB7gl4kMjKAAqHbwLCGeO028AugC+4ki41AQr3ivFAtwojyk8xAMC0rXyWAQEGo50Rb3EsPhSERyNRH02GJQAnIejgARxKIRrwJMKJJPI5KRlOx+McbDQADcUfQmajqWz0FzyBS8eiYXC1FhJkhxtzGWLMZLpbKGVhWRiJQjCEzdDyFZrsdrkbq1Qq6XBSeTlc1VfzCcSLarrXLTTS2A7LfUCDq6Gi3ebPUair71bTOdypRaVWS-QLw8LnarQ+yhfQgya7WHU16ffRQ5w4LgAo4Zd6ICgtHpljczkdAiAft9PhVUn0Mp5GEwZlhSxy4EQDARoUs9qJQK8QlW9vw1tQNo5KOMmCjoRcWLgAMKAqw-McgCeO5b8LnoFohBdLlfPEAAVQACneAKIAJQA+huAIIAZUfDfEIWoAIZhvAhdmQcYrH4dgAEctDQdgQF+cdC0PacQBPNAzysG5AjQAsUBBBZlBgcYGT-EBFxQZdNmve8nzfT8f3IgCgIIaYQLAmUEP4SjQjQYskP3FCp2sGB7GYM4KAvKirwWNdCy3V5mLgQDgNAihwO4ijxj4gS9wPET+DE5oWHIUcQH8ID8MIkAdC0V5i2bbSZJouSJM3bdlNUtiOI0rjIJAGC4IQwSDIgI8LLgYYKAAGTQABrbCLNw6zoXdekyXIyjqNXdzFJ3f8VNY9j1II-zrCC+CrFC4TwrQ8EUDixLHEsvC0A0tLBW5LLL1ckB5I8pSnJYtTOIgiq4FgqrEP02qItoOggWrZKrPagjoV5fIepc6E6Jfd9v1-YaitGvzxp4nT+Jm5DJzq-YkEWhBlta1Lr02pABQ9MinOy2TbwffbGKOwrvJKsatMqkLZtu+aHsQJ7zJetabPeuNU22nLaIBhjDq84rfLK87Asm4Lquh1DrEsJqkt+vq70BC5Bp3JtviAA&tsConfig=N4IgJgpgxgNghgJzgFwJYHsB2IBcyECuEANCAhALboBuEAwuhRRJsgM675GmWrIAi0dEmTCAshGRwwKOJ0IkQEAB4AHCAlTNWcGIKjCUwjngWldMdAHcAygE9WAC0mooggGZwCMZAEkKqsLs8twgbOgECFAQYnCqIYqomFDk2lIwCaRsANaoqgAyqABGdM5Q2QkAvkA

### Repro Code

```typescript
import {Controller, Get} from "@nestjs/common";
import {AppService} from "./app.service";

@Controller()
export class AppController {


    static get BOO(): string {
        return "boo";
    }

    private static get _BAR(): string {
        return "bar";
    }

    private static get _ZAR(): string {
        return "zar";
    }

    set bar(n: string) {
        this._foox = n;
    }

    get bar(): string {
        return this._foox;
    }

    set foo(n: string) {
        this._foox = "public"
    }

    get foo(): string {
        return this._foox;
    }

    private set _foo(n: string) {
        this._foox = "private";
    }

    private get _foo(): string {
        return this._foox;
    }

    private set _goo(n: string) {
        this._foox = "private";
    }

    private get _goo(): string {
        return this._foox;
    }

    private _foox: string = "";

    constructor(private readonly _appService: AppService) {
    }

    @Get()
    getHello(): string {
        return this._appService.getHello();
    }
}
```

### ESLint Config

```javascript
{
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "project": "tsconfig.json",
    "sourceType": "module"
  },
  "plugins": ["@typescript-eslint/eslint-plugin"],
  "extends": [
    "plugin:@typescript-eslint/recommended"
  ],
  "root": true,
  "env": {
    "node": true,
    "jest": true
  },
  "ignorePatterns": [".eslintrc.js"],
  "rules": {
    "@typescript-eslint/interface-name-prefix": "off",
    "@typescript-eslint/explicit-function-return-type": "warn",
    "@typescript-eslint/explicit-module-boundary-types": "off",
    "@typescript-eslint/no-explicit-any": "warn",
    "@typescript-eslint/no-inferrable-types": "off",
    "@typescript-eslint/member-ordering": [
      "error",
      {
        "default": {
          "order": "alphabetically",
          "memberTypes": ["set", "get"]
        },
        "classes": {
          "memberTypes": [
            "public-static-set",
            "public-static-get",
            "protected-static-set",
            "protected-static-get",
            "private-static-set",
            "private-static-get",

            "public-decorated-set",
            "public-decorated-get",
            "public-instance-set",
            "public-instance-get",

            "protected-decorated-set",
            "protected-decorated-get",
            "protected-instance-set",
            "protected-instance-get",

            "private-decorated-set",
            "private-decorated-get",
            "private-instance-set",
            "private-instance-get",


            "field",

            "constructor",

            "method"
          ]
        }
      }
    ],
    "@typescript-eslint/naming-convention": [
      "error",
      {
        "selector": "default",
        "format": ["camelCase"]
      },
      {
        "selector": "variable",
        "modifiers": ["exported"],
        "format": ["UPPER_CASE"],
        "leadingUnderscore": "forbid"
      },
      {
        "selector": "variable",
        "modifiers": [""],
        "format": ["UPPER_CASE"],
        "leadingUnderscore": "require"
      },
      {
        "selector": "parameter",
        "format": ["camelCase"],
        "leadingUnderscore": "forbid"
      },
      {
        "selector": "parameter",
        "modifiers": ["unused"],
        "format": ["camelCase"],
        "leadingUnderscore": "require"
      },
      {
        "selector": "memberLike",
        "modifiers": ["protected"],
        "format": ["camelCase"],
        "leadingUnderscore": "require"
      },
      {
        "selector": "memberLike",
        "modifiers": ["private"],
        "format": ["camelCase"],
        "leadingUnderscore": "require"
      },
      {
        "selector": "accessor",
        "modifiers": ["static"],
        "format": ["UPPER_CASE"],
        "leadingUnderscore": "forbid"
      },
      {
        "selector": "accessor",
        "modifiers": ["static", "protected"],
        "format": ["UPPER_CASE"],
        "leadingUnderscore": "require"
      },
      {
        "selector": "accessor",
        "modifiers": ["static", "private"],
        "format": ["UPPER_CASE"],
        "leadingUnderscore": "require"
      },
      {
        "selector": "typeLike",
        "format": ["PascalCase"]
      }
    ]
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "module": "commonjs",
    "declaration": true,
    "removeComments": true,
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "allowSyntheticDefaultImports": true,
    "target": "es2017",
    "sourceMap": true,
    "outDir": "./dist",
    "baseUrl": "./",
    "incremental": true,
    "skipLibCheck": true
  }
}
```

### Expected Result

I expected no errors at all with the code

### Actual Result

I get 2 errors saying that "Member should be declared before all public instance get definitions".
Maybe it could be related to the naming convention?

### Additional Info

No exceptions found.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.25.0` |
| `@typescript-eslint/parser`            | `5.24.0` |
| `TypeScript`                           | `4.6.2` |
| `ESLint`                               | `8.15.0` |
| `node`                                 | `16.3.2` |

**Comment by bradzacher** [comment]
@armano2 is correct.
based on your current config the correct code would be:
```diff
  export class AppController {
      static get BOO(): string {
          return "boo";
      }
  
      private static get _BAR(): string {
          return "bar";
      }
  
      private static get _ZAR(): string {
          return "zar";
      }
  
      set bar(n: string) {
          this._foox = n;
      }

+     set foo(n: string) {
+         this._foox = "public"
+     }
  
      get bar(): string {
          return this._foox;
      }
  
-     set foo(n: string) {
-         this._foox = "public"
-     }
  
      get foo(): string {
          return this._foox;
      }
  
      private set _foo(n: string) {
          this._foox = "private";
      }
  
+     private set _goo(n: string) {
+         this._foox = "private";
+     }
  
      private get _foo(): string {
          return this._foox;
      }
  
-     private set _goo(n: string) {
-         this._foox = "private";
-     }
  
      private get _goo(): string {
          return this._foox;
      }
  
      private _foox: string = "";
  
      constructor() {
      }
  
      getHello(): string {
          return "Hello";
      }
  }
```

If you want to treat getters and setters the same - you must use the grouped config.

**Comment by KikoCosmetics** [comment]
Hi, I don't want them with the same precedence, I want setters before getters, but also alphabetically
